pynamodb/models.pyi

```diff
@@ -36,14 +39,23 @@ class Model(metaclass=MetaModel):
     def batch_get(cls: Type[_T], items: Iterable[Union[KeyType, Iterable[KeyType]]], consistent_read: Optional[bool] = ..., attributes_to_get: Optional[Sequence[Text]] = ...) -> Iterator[_T]: ...
     @classmethod
     def batch_write(cls: Type[_T], auto_commit: bool = ...) -> BatchWrite[_T]: ...
-    def delete(self, condition: Optional[Any] = ...) -> Any: ...
-    def update(self, actions: List[Any], condition: Optional[Condition] = ...) -> Any: ...
-    def save(self, condition: Optional[Condition] = ...) -> Dict[str, Any]: ...
+    @classmethod
+    def condition_check(cls, hash_key: KeyType, in_transaction: TransactWrite, condition: Any, range_key: Optional[KeyType] = ...): ...
+    @overload
+    def delete(self, condition: Optional[Any] = ..., in_transaction: None = ...) -> Any: ...
+    @overload
+    def delete(self, condition: Optional[Any] = ..., in_transaction: TransactWrite = ...) -> None: ...
+    @overload
+    def update(self, actions: Optional[List[Any]] = ..., condition: Optional[Any] = ..., in_transaction: None = ...) -> Any: ...
+    @overload
+    def update(self, actions: Optional[List[Any]] = ..., condition: Optional[Any] = ..., in_transaction: TransactWrite = ...) -> None: ...
+    @overload
```

Comment:
hello @ikonst im getting this mypy error in the python 3.7 test run and its not very clear to me why its failing (it looks like all the other ones, which are passing), and i was wondering if you knew what was wrong:
```
pynamodb/models.pyi:53: error: Overloaded function signatures 1 and 2 overlap with incompatible return types
```

Reply:
hmpf, we have a pretty’s recent mypy version... I wonder what that that could be.

Reply:
Well, `Any` and `None` are compatible :) But I wish this would work.

Reply:
Maybe try and define the 2nd one without the defaulting?
```python
def save(self, condition: Optional[Any] = ..., in_transaction: TransactWrite) -> None: ...
```

Reply:
that seemed to fix it locally :P

Reply:
added https://github.com/python/mypy/issues/7047